### DISCUSSION TITLE
Chinese expansion board on aliexpress?

### DISCUSSION THREAD

**Discussion by skiphansen** [xjtuecho/EBAZ4205#46] [Q&A]
I found a link to this [expansion board](https://www.aliexpress.com/item/1005001728565885.html) in the Wiki and started to buy one, but then I saw this customer comment:
````
I can't download description & schematic from baidu.com because i have no China phone number 
(I cant register baidu account) So , if You outside China - is useless for You. You have no way get 
schematic & description & code .
````

Which describes me as well.  Does anyone know of a source for the schematic that's more accessible?

Skip

**Comment by DavidJRichards** [comment]
Greetings Skip, I ordered a similar board for my ebaz4205, what actually arrived was slightly different from the photo in the listing. The ebaz torrent includes a schematic and some demo files using the vga output and camera input. Attached is the schematic, mainly the same as the board I have.
[转接板电路图.pdf](https://github.com/xjtuecho/EBAZ4205/files/6148115/default.pdf)
Kind regards, David.

**Reply by skiphansen** [reply]
Thanks David!

**Reply by skiphansen** [reply]
I'm sorry, I'm a bit torrent challenged, what would the URL of the ebaz torrent be?

**Reply by DavidJRichards** [reply]
A member of the Telegram group translated some of the filenames in the torrent:  
The torrent is in the attached zip file, I dont know if it still works.
[FPGA.zip](https://github.com/xjtuecho/EBAZ4205/files/6151545/FPGA.zip)

here is a link to a file describing the contents of the torrent including some translations from Chinese: https://docs.google.com/spreadsheets/d/1gH5yqmZFvx-v_gPzcNBpqH7Yv8eWHvKH8vTq5j1jSfo/edit#gid=837728157

**Comment by firepower9966** [comment]
What are the part numbers of the IC on that board? 
U1, U3, U4

**Comment by DavidJRichards** [comment]
Hi Firepower,

U1 is the video DAC part ADV7123.
There is no footprint for the serial eeprom U2 on my board, if there was then I would expect it to be something like a 24LC16 I2C device.
U3 is a max3232 serial to RS232 converter using 3.3 volt supply and low value capacitors. A sipex SP3232 is fitted on my board - a good alternative.

Kind regards. David.

**Reply by DavidJRichards** [reply]
![IMG_20210316_182758](https://user-images.githubusercontent.com/16197914/111373239-9307ab80-8693-11eb-8255-ccf557bc8cae.jpg)

Picture of the board I received is attached.

btw, there is a 3.3 volt regulator on the board, this is not shown on the schematic. Another change is in the area around pins 17 & 18 of J7. Also the jumpers to select the serial eeprom are not there.

The board is ok, it serves its purpose but could be better thought out.

**Reply by skiphansen** [reply]
Well it has a VGA port which is the main thing I'm interested in for starters.  Thanks for all of the info, I've ordered one.  Hopefully I'll have some time to play when it gets here in a month or so...

**Comment by firepower9966** [comment]
Was interested in buying this but supplier would no provide schematic before purchase. Usuall VGA color bits  is converted with simple resistor ladder DAC. I could not see this and did not know what bit resolution VGA had. Now I know they have dedicated VGA DAC chip using 8 bit color.  This is great info thanks for sharing. https://www.analog.com/media/en/technical-documentation/data-sheets/ADV7123.pdf

**Reply by daybyter** [reply]
You know this project?

http://dossytronics.blogspot.com/2014/03/hdmi-output-from-de0-nano.html

The Zynq should be fast enough to push bits with 250 MHz, so you would not need a VGA DAC?

**Comment by BrainHunter** [comment]
For anyone interested: I just uploaded an example project for the KG4205 board mentioned here. At the moment only the VGA part is implemented in combination with the test pattern generator.
see: https://github.com/BrainHunter/EBAZ4205_with_KG4205

.... anyone has experience in activating framebuffer? ;-)

**Reply by drug490** [reply]
Thank you, it works, but how to output a picture to VGA from memory?

**Reply by daybyter** [reply]
From block ram? From DDR ram? I guess with DDR ram, you will need to cache some data in block ram. Like a raster line of the video mem and maybe the font.
You might also ask in the ebaz 4205 telegram group. Some folks there are doing vga output.

**Reply by langwadt** [reply]
on a different board but anyway. A framebuffer at 0x3fe00000 in DDR RAM, add the framebuffer driver to the kernel and something like this to the devicetree

```
		framebuffer {
			compatible = "simple-framebuffer";
			reg = <0x3FE00000 (800 * 480 * 2)>;
			width = <800>;
			height = <480>;
			stride = <(800 * 2)>;
			format = "r5g6b5";
		};
```

also  added a 'mem=1022M' to the bootargs to prevent the kernel from using the uppe 2M where the fb is

**Reply by drug490** [reply]
![20211027_113856 jpg 22e9773bb975f0f263bfd0505363ee63](https://user-images.githubusercontent.com/62743598/141288313-8eb4db28-f271-45f5-bfce-adb0112c79dc.jpg)
I figured out how to put the image from the Video Test Pattern Generator into AXIS Video DMA (AXI VDMA) and output it to VGA. Now it is interesting to stream the video from the OV7670 camera to VDMA, and then output it to VGA. Maybe there is a ready-made project for EBAZ 4205?

**Reply by BrainHunter** [reply]
> on a different board but anyway. A framebuffer at 0x3fe00000 in DDR RAM, add the framebuffer driver to the kernel and something like this to the devicetree
> 
> ```
> 		framebuffer {
> 			compatible = "simple-framebuffer";
> 			reg = <0x3FE00000 (800 * 480 * 2)>;
> 			width = <800>;
> 			height = <480>;
> 			stride = <(800 * 2)>;
> 			format = "r5g6b5";
> 		};
> ```
> 
> also added a 'mem=1022M' to the bootargs to prevent the kernel from using the uppe 2M where the fb is

After long time i finally tried this and it works!
